src/bindings/c/src/CMakeLists.txt

```diff
@@ -4,29 +4,28 @@
 
 set(TARGET_NAME openvino_c)
 
-file(GLOB SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
-file(GLOB HEADERS ${InferenceEngine_C_API_SOURCE_DIR}/include/*.h)
+file(GLOB SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp ${CMAKE_CURRENT_SOURCE_DIR}/openvino/*.cpp)
+file(GLOB HEADERS ${OpenVINO_C_API_SOURCE_DIR}/include/c_api/*.h ${OpenVINO_C_API_SOURCE_DIR}/include/openvino/*.h)
```

Comment:
Why not?
```suggestion
file(GLOB HEADERS ${OpenVINO_C_API_SOURCE_DIR}/include/*.h)
```

Reply:
It is used to separate legacy api header file and C 2.0 api file, how do you think?

Reply:
But here you generate a set of headers using both paths and it is the same with my suggestion. Looks like it doesn't help to separate headers for the end user.

Reply:
For the end user, we can separate legacy C api  and 2.0 C api in the install package, just like this:


include
├── ie
**│   ├── c_api**
│   ├── cldnn
│   ├── cpp
│   ├── details
│   ├── gna
│   ├── gpu
│   ├── hetero
│   ├── ie_allocator.hpp
│   ├── ie_api.h
│   ├── ie_blob.h
│   ├── ie_common.h
│   ├── ie_compound_blob.h
│   ├── ie_core.hpp
│   ├── ie_data.h
│   ├── ie_extension.h
│   ├── ie_icnn_network.hpp
│   ├── ie_iexecutable_network.hpp
│   ├── ie_iextension.h
│   ├── ie_iinfer_request.hpp
│   ├── ie_input_info.hpp
│   ├── ie_layouts.h
│   ├── ie_locked_memory.hpp
│   ├── ie_parallel.hpp
│   ├── ie_parameter.hpp
│   ├── ie_plugin_config.hpp
│   ├── ie_precision.hpp
│   ├── ie_preprocess.hpp
│   ├── ie_remote_blob.hpp
│   ├── ie_remote_context.hpp
│   ├── ie_transformations.hpp
│   ├── ie_version.hpp
│   ├── inference_engine.hpp
│   ├── multi-device
│   └── vpu
├── ngraph
│   ├── attribute_adapter.hpp
│   ├── attribute_visitor.hpp
│   ├── axis_set.hpp
│   ├── axis_vector.hpp
│   ├── check.hpp
│   ├── compatibility.hpp
│   ├── coordinate_diff.hpp
│   ├── coordinate.hpp
│   ├── deprecated.hpp
│   ├── descriptor
│   ├── dimension.hpp
│   ├── distributed.hpp
│   ├── enum_names.hpp
│   ├── env_util.hpp
│   ├── evaluator.hpp
│   ├── except.hpp
│   ├── factory_adapter.hpp
│   ├── factory.hpp
│   ├── file_util.hpp
│   ├── frontend
│   ├── function.hpp
│   ├── graph_util.hpp
│   ├── interval.hpp
│   ├── log.hpp
│   ├── ngraph.hpp
│   ├── ngraph_visibility.hpp
│   ├── node.hpp
│   ├── node_input.hpp
│   ├── node_output.hpp
│   ├── op
│   ├── opsets
│   ├── ops.hpp
│   ├── output_vector.hpp
│   ├── partial_shape.hpp
│   ├── pass
│   ├── pattern
│   ├── rank.hpp
│   ├── rt_info.hpp
│   ├── runtime
│   ├── shape.hpp
│   ├── shape_util.hpp
│   ├── slice_plan.hpp
│   ├── specialize_function.hpp
│   ├── strides.hpp
│   ├── type
│   ├── type.hpp
│   ├── util.hpp
│   ├── validation_util.hpp
│   ├── variant.hpp
│   ├── version.hpp
│   └── visibility.hpp
└── openvino
    **├── c**
    ├── core
    ├── frontend
    ├── op
    ├── openvino.hpp
    ├── opsets
    ├── pass
    └── runtime

Reply:
But we use this variable only for build. End users use OpenVINOConfig.cmake. They don't work with this script

Reply:
I have tried samples in install directory, we can build c samples after resolve tbb cmake directory issue.

ov2022@ov2022-Z370-AORUS-Gaming-5:~/workspace/capi/openvino/install/samples/c$ ./build_samples.sh -i install -b install

Setting environment variables for building samples...
[setupvars.sh] OpenVINO environment initialized
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- zlib (1.2.11) is found at /usr
CMake Warning at common/opencv_c_wrapper/CMakeLists.txt:17 (message):
  OPENCV is disabled or not found, opencv_c_wrapper is built without OPENCV
  support


-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ov2022/workspace/capi/openvino/install/samples/c/install
Scanning dependencies of target opencv_c_wrapper
Scanning dependencies of target hello_nv12_input_classification_c
[ 42%] Building C object common/opencv_c_wrapper/CMakeFiles/opencv_c_wrapper.dir/bmp_reader.c.o
[ 42%] Building CXX object common/opencv_c_wrapper/CMakeFiles/opencv_c_wrapper.dir/opencv_c_wrapper.cpp.o
[ 42%] Building C object hello_nv12_input_classification/CMakeFiles/hello_nv12_input_classification_c.dir/main.c.o
[ 57%] Linking C executable ../intel64/Release/hello_nv12_input_classification_c
[ 71%] Linking CXX shared library ../../intel64/Release/lib/libopencv_c_wrapper.so
[ 71%] Built target opencv_c_wrapper
Scanning dependencies of target hello_classification_c
[ 85%] Building C object hello_classification/CMakeFiles/hello_classification_c.dir/main.c.o
[100%] Linking C executable ../intel64/Release/hello_classification_c
[100%] Built target hello_nv12_input_classification_c
[100%] Built target hello_classification_c
Scanning dependencies of target ie_samples
[100%] Built target ie_samples
-- Install configuration: "Release"
-- Install component: "samples_bin"
-- Installing: /home/ov2022/workspace/capi/openvino/install/samples/c/install/install/samples_bin/libopencv_c_wrapper.so
-- Installing: /home/ov2022/workspace/capi/openvino/install/samples/c/install/install/samples_bin/hello_classification_c
-- Set runtime path of "install/samples_bin/hello_classification_c" to ""
-- Installing: /home/ov2022/workspace/capi/openvino/install/samples/c/install/install/samples_bin/hello_nv12_input_classification_c
-- Set runtime path of "install/samples_bin/hello_nv12_input_classification_c" to ""

Build completed, you can find binaries for all samples in the install/samples_bin subfolder.

Reply:
This variable defines headers only for the build process. So it doesn't help us to split old and new headers for the end user. We can use one path as I proposed.

In order to have a difference after the installation we need to use the same approach as for OV C++ API.
Legacy config works with inference_engine target, the new config works with openvino target and these targets define different include directories.

Reply:
ok

Reply:
done